lib/buffer.js

```diff
@@ -17,6 +17,48 @@ var poolSize, poolOffset, allocPool;
 
 
 binding.setupBufferJS(Buffer.prototype, bindingObj);
+
+const swap16n = binding.swap16;
+const swap32n = binding.swap32;
+
+function swap(b, n, m) {
+  const i = b[n];
+  b[n] = b[m];
+  b[m] = i;
+}
+
+Buffer.prototype.swap16 = function swap16() {
+  // For Buffer.length < 512, it's generally faster to
+  // do the swap in javascript. For larger buffers,
+  // dropping down to the native code is faster.
+  const len = this.length;
+  if (len % 2 !== 0)
+    throw new RangeError('Buffer length must be a multiple of 16-bits');
```

Comment:
Incredible nitpicking, but nonetheless I would opt for one of these:

`Buffer size must be a multiple of 16 bits`

or:

`Buffer length must be a multiple of 2`